File: code_samples/kotlin/oop_polymorphism.kt
/**
 * Oop: Polymorphism
 * AI/ML Training Sample
 */

class Polymorphism {
    var data: String = ""
        private set
    
    fun process(input: String) {
        data = input
    }
    
    fun getData(): String = data
    
    fun validate(): Boolean = data.isNotEmpty()
}

fun main() {
    val instance = Polymorphism()
    instance.process("example")
    println("Data: ${instance.getData()}")
    println("Valid: ${instance.validate()}")
}

File: code_samples/dart/oop_polymorphism.dart
// Oop: Polymorphism
// AI/ML Training Sample

class Polymorphism {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Polymorphism();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
